README.md
# ELS - Enterprise Linux/EuroLinux Sources - libabw
 
## Usage:
  Checkout branch or tag. Use `make sources` to recreate  SOURCE0 archive.


src/ChangeLog
2014-12-20  David Tardon  <dtardon@redhat.com>  [0822e285c7acda1e4457aa9a60e9d69dafce9f80]

	prepare for a release

	Change-Id: I7c65bc7736fb18ae1ee0be8cb059c942bda52389

2014-12-20  David Tardon  <dtardon@redhat.com>  [52821af0b9dcd4f0856f2952e54bc64e0d9342a5]

	fix copypasta

	Change-Id: I53bbd861af92e833f20333fc6db690b482bc313c

2014-12-20  David Tardon  <dtardon@redhat.com>  [12bf83bf39fb11151c14c2615806eb9fa2c46c02]

	fix processing of mixed paragraph content

	I would much prefer to do this in the parser, but calling closeSpan()
	unconditionally just seems wrong and I do not want to introduce a new
	function to ABWCollector just to be able to check if a span is opened.

	Change-Id: Icf6950a5bbc8d401f5773270d23a45ddf2e41dcc

2014-12-20  David Tardon  <dtardon@redhat.com>  [1f4ef02f77320dfd4de3db6c635c4f9935cd3fa8]

	fix condition

	Change-Id: I359a44e56ffecaa6d01b561a312cea9b3c1a4575

2014-12-20  David Tardon  <dtardon@redhat.com>  [6f8f69d5e5d2511f21dc6014628f4abb6087a9ad]

	print debug messages to stderr

	Change-Id: Ic70e939cfe68a81165418482c2dc9e07473ef0b7

2014-12-19  David Tardon  <dtardon@redhat.com>  [4712f04685caf87c48d448ff14ad0a6341cc1e62]

	import headings

	Change-Id: I49e1ce400fe1b3162170f357400a98cba182fd66

2014-12-19  David Tardon  <dtardon@redhat.com>  [eb2aab1972a7239488ddc86ce1c73c1124ab68ba]

	document props are global

	Change-Id: I4c4c2d56fcd4c616336d37e0826430e132887151

2014-12-19  David Tardon  <dtardon@redhat.com>  [8f0d36795e809d47113e20eac718b29675801614]

	import metadata

	Change-Id: I03f210616f7d2a9d5e24220b95a0334e63760a5a

2014-12-19  David Tardon  <dtardon@redhat.com>  [78ac2ee4cebbc7755cfda36851af9a9e0f9bbc96]

	set writing mode from document default

	Change-Id: Ia04d00310cb73d6af6bb19cb21c4ec52011d5969

2014-12-19  David Tardon  <dtardon@redhat.com>  [92e48a98cb470e8dc76c6873bc30198a36f81d77]

	eliminate some copypasta

	Change-Id: I295b98876b19d514d114d115ef8ce8e7ce00d6d1

2014-12-19  David Tardon  <dtardon@redhat.com>  [b2290b8a30bd9e12bb4d8ab2079a25e8011d1258]

	use typedef for property map

	Change-Id: Ia29e9fa6084696105783fa09f92738f2c5349496

2014-12-19  David Tardon  <dtardon@redhat.com>  [93baf6994778311b3ffd8afd2e98dfa15389e737]

	add support for text language

	Change-Id: I4e184195c3e54d54a6627aec110fc63d250fb211

2014-11-24  David Tardon  <dtardon@redhat.com>  [acafe5bf1dc491edab3980a7bb3a91f05fd48e03]

	add compile script to .gitignore

	Change-Id: I0fbd5cfb63031b89c48dee9f1634c3422ef0cd33

2014-11-24  David Tardon  <dtardon@redhat.com>  [d80d7c7730a4599613552e69ac65e9ec76525fcb]

	only export public symbols


2014-06-16  Michael Stahl  <mstahl@redhat.com>  [cba9f861292cd3e1a3c2e5baaccf8f87a42cfc34]

	libabw: avoid some quoting issue with Win32 make


2014-05-23  David Tardon  <dtardon@redhat.com>  [3c0b37ba5d225ea71f42c3179f4438b12ec5abcb]

	prepare for a new release

	Change-Id: I34cbd95ad975afdfc8892c38ebe06c90030b37fb

2014-05-23  David Tardon  <dtardon@redhat.com>  [51e28e34f0bfb5e4a0eff2e444add3f62b796a23]

	put the newest version on top

	Change-Id: Icb44c9efd0246b4c17334b357c1454bd60a53dc5

2014-04-16  David Tardon  <dtardon@redhat.com>  [f43bb981cab1df2daa5823a3d7d53c54170cca92]

	add other used projects to .pc file

	Change-Id: I7dfe0a26b53857b9ac7625ff973540bc3eeaf08a

2014-04-16  David Tardon  <dtardon@redhat.com>  [2d459731222782fcfc0833698c4acc74ba2cf597]

	require librevenge-stream too

	Change-Id: I2bc0ebad7e4734a6fc42a7d0fae9755d07f77600

2014-02-10  David Tardon  <dtardon@redhat.com>  [0bcf974769248d8e7b5b35733d2c655653585903]

	drop stray files

	Change-Id: I45260ebcee40ce2b12196bf95dda573b1ce38a49

2014-01-29  David Tardon  <dtardon@redhat.com>  [f16e6cf59bdc681d410cd356e4eff89abd98e372]

	CID#1165311 missing break

	There is no need to set ret if we are going to return immediately
	anyway.

	Change-Id: I390a4f67c907d75c554673f976b952e1008c94ae

2014-01-29  David Tardon  <dtardon@redhat.com>  [f3723f79a20de230b3159b8d73debb567bb520f4]

	add myself to contributors

	Change-Id: I8fea4c66e58690ef808b8ae48bcb1e3186deb07d

2014-01-23  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [1b941a8b1e2aa1909c1f6a0d2c8aeeb04f8ec4b2]

	Ignore cells if not inside the <table></table> tags

	Change-Id: I5dfc2d05a6191f9cbf97d300a9ca1624b40e21e9

2014-01-21  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [393f2b5bb8bbb802526771baf1087725cf920452]

	std::string constructor with null string might crash in some cases

	Change-Id: Ie692ffbece7235fe32abf46e84d4f2a29442e96f

2014-01-17  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [bd1b2b63526efecc3d57c12a801b14dae7a3f47a]

	Put as much as you can into the try block

	(cherry picked from commit 095b595a81cc18f3f16960c202079d2e961d1634)

	Conflicts:
		src/lib/AbiDocument.cpp

	Change-Id: Id9c4193228bc7af7da685edb0db07c91f6c8de24

2014-01-17  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [0c16f59fe7a3aed7cc055a81facbbc1fb35998d7]

	All invocation of xmlReaderForStream use the same options

	(cherry picked from commit e68f8f29ec365cc07edf6f1b1dc15c7f75f40051)

	Conflicts:
		src/lib/ABWXMLHelper.cpp
		src/lib/ABWXMLHelper.h
		src/lib/AbiDocument.cpp

	Change-Id: I6f61c92db22114e3592cd5d24516bcd879422566

2014-01-16  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [720f251278f53d3066670eef60eb2611aa95bbbc]

	No need to use the stuff from inttypes here

	(cherry picked from commit 2b21394e42767ece602741538e485c2ec3cb5c85)

	Conflicts:
		src/lib/ABWStylesCollector.cpp

	Change-Id: Ia8087ed0b18c8ed12ec75d272bf8f409038f33b0

2014-01-15  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [87e73a202a1ea96fbf361f32e71ad8efc42367d0]

	Strip first tab in a list text

	Change-Id: I46a4d0f36b2d255a30c669f24dabe65852c35d61

2014-01-15  Fridrich Štrba  <fridrich.strba@bluewin.ch>  [902454110899751868ba9ab56f0ef554f8239eb5]

	Some NEWS

	Change-Id: I47e4a37628da83d467bbe757efe6f336b30cbe32
	(cherry picked from commit 9817ef41ad922387d56a0a8c5abb862977f33dea)


src/src/lib/tokens.h
#ifndef __ABWXMLTOKENS_HXX__
#define __ABWXMLTOKENS_HXX__

const int XML_A = 1;
const int XML_ABIWORD = 2;
const int XML_AWML = 3;
const int XML_BOOKMARK = 4;
const int XML_BR = 5;
const int XML_C = 6;
const int XML_CBR = 7;
const int XML_CELL = 8;
const int XML_D = 9;
const int XML_DATA = 10;
const int XML_ENDNOTE = 11;
const int XML_F = 12;
const int XML_FIELD = 13;
const int XML_FOOT = 14;
const int XML_FRAME = 15;
const int XML_HISTORY = 16;
const int XML_I = 17;
const int XML_IGNOREDWORDS = 18;
const int XML_IMAGE = 19;
const int XML_IW = 20;
const int XML_L = 21;
const int XML_LISTS = 22;
const int XML_M = 23;
const int XML_METADATA = 24;
const int XML_P = 25;
const int XML_PAGESIZE = 26;
const int XML_PBR = 27;
const int XML_REVISIONS = 28;
const int XML_S = 29;
const int XML_SECTION = 30;
const int XML_STYLES = 31;
const int XML_TABLE = 32;
const int XML_VERSION = 33;

const int XML_TOKEN_COUNT = 33;

const int XML_TOKEN_INVALID = -1;

#endif


src/src/lib/tokenhash.h
/* C++ code produced by gperf version 3.0.4 */
/* Command-line: /usr/bin/gperf --compare-strncmp -C -m 20 ../../src/lib/tokens.gperf  */
/* Computed positions: -k'1-2' */

#if !((' ' == 32) && ('!' == 33) && ('"' == 34) && ('#' == 35) \
      && ('%' == 37) && ('&' == 38) && ('\'' == 39) && ('(' == 40) \
      && (')' == 41) && ('*' == 42) && ('+' == 43) && (',' == 44) \
      && ('-' == 45) && ('.' == 46) && ('/' == 47) && ('0' == 48) \
      && ('1' == 49) && ('2' == 50) && ('3' == 51) && ('4' == 52) \
      && ('5' == 53) && ('6' == 54) && ('7' == 55) && ('8' == 56) \
      && ('9' == 57) && (':' == 58) && (';' == 59) && ('<' == 60) \
      && ('=' == 61) && ('>' == 62) && ('?' == 63) && ('A' == 65) \
      && ('B' == 66) && ('C' == 67) && ('D' == 68) && ('E' == 69) \
      && ('F' == 70) && ('G' == 71) && ('H' == 72) && ('I' == 73) \
      && ('J' == 74) && ('K' == 75) && ('L' == 76) && ('M' == 77) \
      && ('N' == 78) && ('O' == 79) && ('P' == 80) && ('Q' == 81) \
      && ('R' == 82) && ('S' == 83) && ('T' == 84) && ('U' == 85) \
      && ('V' == 86) && ('W' == 87) && ('X' == 88) && ('Y' == 89) \
      && ('Z' == 90) && ('[' == 91) && ('\\' == 92) && (']' == 93) \
      && ('^' == 94) && ('_' == 95) && ('a' == 97) && ('b' == 98) \
      && ('c' == 99) && ('d' == 100) && ('e' == 101) && ('f' == 102) \
      && ('g' == 103) && ('h' == 104) && ('i' == 105) && ('j' == 106) \
      && ('k' == 107) && ('l' == 108) && ('m' == 109) && ('n' == 110) \
      && ('o' == 111) && ('p' == 112) && ('q' == 113) && ('r' == 114) \
      && ('s' == 115) && ('t' == 116) && ('u' == 117) && ('v' == 118) \
      && ('w' == 119) && ('x' == 120) && ('y' == 121) && ('z' == 122) \
      && ('{' == 123) && ('|' == 124) && ('}' == 125) && ('~' == 126))
/* The character set is not based on ISO-646.  */
#error "gperf generated tables don't work with this execution character set. Please report a bug to <bug-gnu-gperf@gnu.org>."
#endif

#line 5 "../../src/lib/tokens.gperf"
struct xmltoken
{
  const char *name;
  int tokenId;
};

#define TOTAL_KEYWORDS 33
#define MIN_WORD_LENGTH 1
#define MAX_WORD_LENGTH 12
#define MIN_HASH_VALUE 1
#define MAX_HASH_VALUE 33
/* maximum key range = 33, duplicates = 0 */

class Perfect_Hash
{
private:
  static inline unsigned int hash(const char *str, unsigned int len);
public:
  static const struct xmltoken *in_word_set(const char *str, unsigned int len);
};

inline unsigned int
Perfect_Hash::hash(register const char *str, register unsigned int len)
{
  static const unsigned char asso_values[] =
  {
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34,  1,  0, 14,
    27,  3,  5, 21, 23,  0, 34, 34, 26,  8,
    19, 17,  2, 34,  2,  6, 12, 34, 12, 18,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
    34, 34, 34, 34, 34, 34
  };
  register int hval = len;

  switch (hval)
  {
  default:
    hval += asso_values[(unsigned char)str[1]];
  /*FALLTHROUGH*/
  case 1:
    hval += asso_values[(unsigned char)str[0]];
    break;
  }
  return hval;
}

static const struct xmltoken wordlist[] =
{
  {(char *)0, 0},
#line 27 "../../src/lib/tokens.gperf"
  {"i",XML_I},
#line 11 "../../src/lib/tokens.gperf"
  {"a",XML_A},
#line 35 "../../src/lib/tokens.gperf"
  {"p",XML_P},
#line 15 "../../src/lib/tokens.gperf"
  {"br",XML_BR},
#line 37 "../../src/lib/tokens.gperf"
  {"pbr",XML_PBR},
#line 22 "../../src/lib/tokens.gperf"
  {"f",XML_F},
#line 39 "../../src/lib/tokens.gperf"
  {"s",XML_S},
#line 12 "../../src/lib/tokens.gperf"
  {"abiword",XML_ABIWORD},
#line 33 "../../src/lib/tokens.gperf"
  {"m",XML_M},
#line 23 "../../src/lib/tokens.gperf"
  {"field",XML_FIELD},
#line 36 "../../src/lib/tokens.gperf"
  {"pagesize",XML_PAGESIZE},
#line 25 "../../src/lib/tokens.gperf"
  {"frame",XML_FRAME},
#line 29 "../../src/lib/tokens.gperf"
  {"image",XML_IMAGE},
#line 38 "../../src/lib/tokens.gperf"
  {"revisions",XML_REVISIONS},
#line 16 "../../src/lib/tokens.gperf"
  {"c",XML_C},
#line 40 "../../src/lib/tokens.gperf"
  {"section",XML_SECTION},
#line 17 "../../src/lib/tokens.gperf"
  {"cbr",XML_CBR},
#line 42 "../../src/lib/tokens.gperf"
  {"table",XML_TABLE},
#line 34 "../../src/lib/tokens.gperf"
  {"metadata",XML_METADATA},
#line 30 "../../src/lib/tokens.gperf"
  {"iw",XML_IW},
#line 18 "../../src/lib/tokens.gperf"
  {"cell",XML_CELL},
#line 43 "../../src/lib/tokens.gperf"
  {"version",XML_VERSION},
#line 13 "../../src/lib/tokens.gperf"
  {"awml",XML_AWML},
#line 41 "../../src/lib/tokens.gperf"
  {"styles",XML_STYLES},
#line 14 "../../src/lib/tokens.gperf"
  {"bookmark",XML_BOOKMARK},
#line 24 "../../src/lib/tokens.gperf"
  {"foot",XML_FOOT},
#line 31 "../../src/lib/tokens.gperf"
  {"l",XML_L},
#line 19 "../../src/lib/tokens.gperf"
  {"d",XML_D},
#line 21 "../../src/lib/tokens.gperf"
  {"endnote",XML_ENDNOTE},
#line 26 "../../src/lib/tokens.gperf"
  {"history",XML_HISTORY},
#line 32 "../../src/lib/tokens.gperf"
  {"lists",XML_LISTS},
#line 20 "../../src/lib/tokens.gperf"
  {"data",XML_DATA},
#line 28 "../../src/lib/tokens.gperf"
  {"ignoredwords",XML_IGNOREDWORDS}
};

const struct xmltoken *
Perfect_Hash::in_word_set(register const char *str, register unsigned int len)
{
  if (len <= MAX_WORD_LENGTH && len >= MIN_WORD_LENGTH)
  {
    register int key = hash(str, len);

    if (key <= MAX_HASH_VALUE && key >= 0)
    {
      register const char *s = wordlist[key].name;

      if (s && *str == *s && !strncmp(str + 1, s + 1, len - 1) && s[len] == '\0')
        return &wordlist[key];
    }
  }
  return 0;
}
#line 44 "../../src/lib/tokens.gperf"

